### ISSUE TITLE
Exception when combining families on merge template with a large dataset

### ISSUE THREAD

**Issue by chrisfunk** [SparkDevNetwork/Rock#3840]
### Description
When generating a Merge Template with "Combine Family Members" selected from a large dataset (5400+) a SQL Exception will occur. 
![MailMerge exception screenshot](https://user-images.githubusercontent.com/2363131/62486141-11942b80-b784-11e9-8e41-8e4be2f118d2.png)

### Steps to Reproduce
1. Load a person report with a large number of people (~5400+)
1. Click on the grid's "Merge records into Merge Template" button
1. Select the "Combine Family Members" checkbox on the Merge Template form
1. Select a merge template (I used Envelopes, Labels... > 5160 lables)
1. Select the "Merge" button

**Expected behavior:**

A merge document will be generated 

**Actual behavior:**

The SQL Exception occurs

### Versions
* **Rock Version:** 1.8.8 (also confirmed in 1.9.0)
* **Client Culture Setting:** en-us

**Comment by nairdo** [comment]
I was able to reproduce with a report containing 14,000 people.  Based on the exception, it would seem there is currently a limit when extremely complex, large queries are involved:

```
-- SqlException in .Net SqlClient Data Provider
Message
The query processor ran out of internal resources and could not produce a query plan. 
This is a rare event and only expected for extremely complex queries or queries that 
reference a very large number of tables or partitions. Please simplify the query. If you 
believe you have received this message in error, contact Customer Support Services 
for more information.
```

**Comment by cabal95** [comment]
Man, I so want to see the query it generated to get that error.

**Comment by nairdo** [comment]
@chrisfunk  If you want to try and test a possible solution in your test environment, you might want to try removing the ToList() on the https://github.com/SparkDevNetwork/Rock/blob/develop/RockWeb/Blocks/Core/MergeTemplateEntry.ascx.cs#L284

```CSharp
var orderIds = qryPersons.Select( a => a.Id );
```

...and then `delete` this line:

```CSharp
mergeObjectsDictionary = mergeObjectsDictionary.OrderBy( a => orderIds.IndexOf( a.Key ) ).ToDictionary( x => x.Key, y => y.Value );
```

I was able to get it to work on over 14,000 combined people records on a test system.

**Comment by chrisfunk** [comment]
@nairdo tried this without issues. Thanks!

**Comment by adamhann** [comment]
We are having this exact same issues. Is there a fix on the way?

**Comment by notGerryMiller** [comment]
@nairdo Thanks! That worked for us.  Does anyone know if this is going to get patched?

**Comment by nairdo** [comment]
@notGerryMiller Not Gerry, This will be addressed in v9.5 and/or v10.2

**Comment by notGerryMiller** [comment]
@nairdo Thanks for the update.  This was a time vampire for me on Friday.
